src/draw/nxp/vglite/lv_draw_vglite.c

```diff
@@ -442,41 +478,57 @@ static void _vglite_execute_drawing(lv_draw_vglite_unit_t * u)
 }
 
 #if LV_USE_VGLITE_DRAW_ASYNC
-static inline void _vglite_queue_task(lv_draw_task_t * task_act)
+static inline void _vglite_queue_task(lv_draw_task_t * task)
 {
-    _draw_task_buf[_tail].task = task_act;
+    VGLITE_ASSERT_MSG(((_tail + 1) % VGLITE_TASK_BUF_SIZE) != _head, "VGLite task buffer full.");
+
+    _draw_task_buf[_tail].task = task;
     _draw_task_buf[_tail].flushed = false;
     _tail = (_tail + 1) % VGLITE_TASK_BUF_SIZE;
 }
 
-static inline void _vglite_signal_task_ready(lv_draw_task_t * task_act)
+static inline void _vglite_signal_task_ready(lv_draw_task_t * task)
+{
+    /* Signal the ready state to dispatcher. */
+    task->state = LV_DRAW_TASK_STATE_READY;
+    _head = (_head + 1) % VGLITE_TASK_BUF_SIZE;
+
+    /* No need to cleanup the tasks in buffer as we advance with the _head. */
+}
+
+static inline void _vglite_signal_all_task_ready(void)
```

Comment:
Please forward declare the new static functions.

Reply:
Is this mandatory?
Those small static functions (helpers) are implemented before their use, just to have the code cleaner.

All the callback functions used statically like `_vglite_dispatch`, `_vglite_delete`, `_vglite_execute`.... are declared on top of the file.

Reply:
In these very specific files I'm not too picky about it, but we should follow `lv_templ.c/h` where the static functions are moved at the bottom.

We can leave them here now.

Reply:
The static functions are all at the buttom of the file. `_vglite_signal_all_task_ready` is a static function called by another static function :) This is why I called it helper and marked as inline.